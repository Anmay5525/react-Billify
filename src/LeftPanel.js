const LeftPanel = ({ route, handleRouteChange }) => {
  return (
    <div class="left-panel">
      <div
        class={`btn ${route === "Customers" ? "active" : ""}`}
        id="Customers"
        onClick={() => handleRouteChange("Customers")}
      >
        Customers
      </div>

      <div
        class={`btn ${route === "Items" ? "active" : ""}`}
        id="Items"
        onClick={() => handleRouteChange("Items")}
      >
        Items
      </div>
      <div
        class={`btn ${route === "Invoices" ? "active" : ""}`}
        id="Invoices"
        onClick={() => handleRouteChange("Invoices")}
      >
        Invoices
      </div>
    </div>
  );
};

export default LeftPanel;
