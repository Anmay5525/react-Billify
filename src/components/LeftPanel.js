import React from "react";
import { Link, useLocation } from "react-router-dom";

const LeftPanel = () => {
  const location = useLocation();
  const activeBtn = location.pathname.split("/")[1];
  return (
    <div className="left-panel">
      <Link to="/Customers">
        <button
          type="button"
          className={`btn ${activeBtn === "Customers" ? "active" : ""}`}
          id="Customers"
        >
          {/* change to see if CI stops if test fails */}
          Customers
        </button>
      </Link>
      <Link to="/Items">
        <button
          type="button"
          className={`btn ${activeBtn === "Items" ? "active" : ""}`}
          id="Items"
        >
          Items
        </button>
      </Link>
      <Link to="/Invoices">
        <button
          type="button"
          className={`btn ${activeBtn === "Invoices" ? "active" : ""}`}
          id="Invoices"
        >
          Invoices
        </button>
      </Link>
    </div>
  );
};

export default LeftPanel;
