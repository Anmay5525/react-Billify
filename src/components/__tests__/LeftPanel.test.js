import React from 'react';
import ReactDOM from 'react-dom';
import { mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import LeftPanel from '../LeftPanel';
import { BrowserRouter as Router, MemoryRouter} from "react-router-dom";

describe('Testing LeftPanel component if it', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><LeftPanel></LeftPanel></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('makes Customers active', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/Customers']}>
      <LeftPanel></LeftPanel>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('makes Items active', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/Items']}>
      <LeftPanel></LeftPanel>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('makes Invoices active', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/Invoices']}>
      <LeftPanel></LeftPanel>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});