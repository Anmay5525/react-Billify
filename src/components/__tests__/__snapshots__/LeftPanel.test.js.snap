// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LeftPanel component if it makes Customers active 1`] = `
<div
  className="left-panel"
>
  <a
    href="/Customers"
    onClick={[Function]}
  >
    <button
      className="btn active"
      id="Customers"
      type="button"
    >
      Customers
    </button>
  </a>
  <a
    href="/Items"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Items"
      type="button"
    >
      Items
    </button>
  </a>
  <a
    href="/Invoices"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Invoices"
      type="button"
    >
      Invoices
    </button>
  </a>
</div>
`;

exports[`Testing LeftPanel component if it makes Invoices active 1`] = `
<div
  className="left-panel"
>
  <a
    href="/Customers"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Customers"
      type="button"
    >
      Customers
    </button>
  </a>
  <a
    href="/Items"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Items"
      type="button"
    >
      Items
    </button>
  </a>
  <a
    href="/Invoices"
    onClick={[Function]}
  >
    <button
      className="btn active"
      id="Invoices"
      type="button"
    >
      Invoices
    </button>
  </a>
</div>
`;

exports[`Testing LeftPanel component if it makes Items active 1`] = `
<div
  className="left-panel"
>
  <a
    href="/Customers"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Customers"
      type="button"
    >
      Customers
    </button>
  </a>
  <a
    href="/Items"
    onClick={[Function]}
  >
    <button
      className="btn active"
      id="Items"
      type="button"
    >
      Items
    </button>
  </a>
  <a
    href="/Invoices"
    onClick={[Function]}
  >
    <button
      className="btn "
      id="Invoices"
      type="button"
    >
      Invoices
    </button>
  </a>
</div>
`;
